### PR TITLE
Git Hooks for push

### PR DESCRIPTION
**Pull request by mcortesi**
### Description

Prevent pushing old files; or unencoded mnemonics

### Tested

local testing with git

### CODE REVIEW

File: scripts/hooks/pre-push.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  })
+
+for (const change of changes) {
+  // console.log('Checking commits to push to ', change.remoteRef)
+  const [from, to] = getCommitRange(change, remoteName)
+
+  const changedFiles = getChangedFiled(from, to)
+
+  const pushedMnemonicFiles = changedFiles.filter(
+    (name) => path.basename(name).startsWith('.env.mnemonic') && path.extname(name) !== 'enc'
```
Comment (ashishb) [inline scripts/hooks/pre-push.js]: Nit: Might change `enc` to `.enc` to catch `.env.mnemonic.some-name-which-ends-in-enc`

File: scripts/hooks/pre-push.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  )
+  if (pushedMnemonicFiles.length > 0) {
+    console.error(`Trying to push conflicting files`)
+    console.log(`Conflicting Files:\n  ${pushedMnemonicFiles.join('\n  ')}`)
+    console.error(chalk.red(`(${change.remoteRef}) Push rejected!`))
+    process.exit(1)
+  }
+
+  const firstCommitDate = getDateFromFirstCommit(from, to)
+  if (firstCommitDate <= THRESHOLD_DATE) {
```
Comment (ashishb) [inline scripts/hooks/pre-push.js]: Nit: Commit created on `THRESHOLD_DATE` should be fine. So `<` might be more appropriate.

**Review by ashishb — DISMISSED** [review]

File: scripts/hooks/pre-push.js (hunk 1 above)
Comment (mcortesi) [reply scripts/hooks/pre-push.js]: good catch it's actually buggy... I **must** use the `.`

From node console
```
> path.extname('packages/mobile/package.json')
'.json'
```

**Review by ashishb — DISMISSED** [review]

**Review by isabellewei — DISMISSED** [review]
Looks good! I think you need to commit your `yarn.lock`

**Review by ashishb — DISMISSED** [review]

**Review by ashishb — APPROVED** [review]

**Pull request merged**